ReportingServicesTools/Functions/CatalogItems/Set-RsItemDataSource.ps1

```diff
@@ -0,0 +1,177 @@
+            $dataSources[0].Item.Username = 'domain\\user'
+            $dataSources[0].Item.Password = 'UserPassword'
+            $dataSources[0].Item.WindowsCredentials = $true # (optional) Setting this property would make Report Server treat these credentials as Windows Credentials. By default, Report Server treats credentials as SQL Credentials.
+            $dataSources[0].Item.ImpersonateUser = $true # (optional) Setting this property would make Report Server connect using the specified credentials but fetch data using the credentials of the user who is using this data source via Report/Shared Dataset.
+            Set-RsItemDataSource -RsItem '/report' -DataSource $dataSources
+
+            Description
+            -----------
+            This command will update 1st data source associated to '/report' catalog item at the Report Server located at http://localhost/reportserver. It will instruct Report Server to use stored credentials when fetching data.
+
+        .EXAMPLE
+            $dataSources = Get-RsItemDataSource -RsItem '/report'
+            $dataSources[0].Item.CredentialRetrieval = 'Prompt'
+            $dataSources[0].Item.Prompt = 'Please enter your credentials.' # (optional) Enter the message you want to display to user when asking for their credentials
+            $dataSources[0].Item.WindowsCredentials = $true # (optional) Setting this property would make Report Server treat these credentials as Windows Credentials. By default, Report Server treats credentials as SQL Credentials.
+            $dataSources[0].Item.Username = '' # (optional) clear up username if you had previously specified one.
+            $dataSources[0].Item.Password = '' # (optional) clear up password if you had previously specified one.
+            Set-RsItemDataSource -RsItem '/report' -DataSource $dataSources
+
+            Description
+            -----------
+            This command will update 1st data source associated to '/report' catalog item at the Report Server located at http://localhost/reportserver. It will instruct Report Server to ask user for credentials when fetching data.
+
+        .EXAMPLE
+            $dataSources = Get-RsItemDataSource -RsItem '/report'
+            $dataSources[0].Item.CredentialRetrieval = 'None'
+            $dataSources[0].Item.Username = '' # (optional) clear up username if you had previously specified one.
+            $dataSources[0].Item.Password = '' # (optional) clear up password if you had previously specified one.
+            Set-RsItemDataSource -RsItem '/report' -DataSource $dataSources
+
+            Description
+            -----------
+            This command will update 1st data source associated to '/report' catalog item at the Report Server located at http://localhost/reportserver. It will instruct Report Server to specify no credentials when fetching data.
+    #>
+
+    [cmdletbinding()]
+    param
+    (
+        [Alias('ItemPath', 'DataSourcePath', 'Path')]
```

Comment:
do you need alias?

Reply:
Added these for consistency with other commands that use the SOAP API (e.g. Set-RsDataSource which is for updating Shared Data Sources).